src/runtime/graph/graph_runtime_factory.h

```diff
@@ -0,0 +1,155 @@
+
+  /*!
+   * \brief Get member function to front-end
+   * \param name The name of the function.
+   * \param sptr_to_self The pointer to the module node.
+   * \return The corresponding member function.
+   */
+  PackedFunc GetFunction(const std::string& name, const ObjectPtr<Object>& sptr_to_self) final;
+
+  /*!
+   * \return The type key of the executor.
+   */
+  const char* type_key() const override { return "GraphRuntimeFactory"; }
+
+  /*!
+   * \brief Save the module to binary stream.
+   * \param stream The binary stream to save to.
+   */
+  void SaveToBinary(dmlc::Stream* stream) override;
+
+  /*!
+   * \brief Create a specific runtime module
+   * \param module The module we will be used for creating runtime
+   * \param ctxs The context of the host and devices where graph nodes will be
+   *  executed on.
+   * \return created runtime module
+   */
+  Module RuntimeCreate(Module module, const std::vector<TVMContext>& ctxs);
+
+  /*!
+   * \brief Create a specific debug runtime module
+   * \param module The module we will be used for creating runtime
+   * \param ctxs The context of the host and devices where graph nodes will be
+   *  executed on.
+   * \return created debug runtime module
+   */
+  Module DebugRuntimeCreate(Module module, const std::vector<TVMContext>& ctxs);
+
+  /*!
+   * \brief Select the specific module
+   * \param name The name of the module
+   * \return selected module
+   */
+  Module SelectModule(const std::string& name);
+
+  /*!
+   * \brief Set params.
+   * \param graph_runtime The graph runtime we want to set the params into.
+   * \param params The graph params value we want to set.
+   */
+  void SetParams(GraphRuntime* graph_runtime,
+                 const tvm::Map<String, tvm::runtime::NDArray>& params) const {
+    tvm::Map<String, tvm::runtime::NDArray> value = params;
+    // upload big arrays first to avoid memory issue in rpc mode
+    std::vector<std::string> keys;
+    for (const auto& p : value) {
+      keys.emplace_back(p.first);
+    }
+    std::sort(std::begin(keys), std::end(keys),
+              [&](const std::string& lhs, const std::string& rhs) -> bool {
+                auto lhs_shape = value[lhs].Shape();
+                auto rhs_shape = value[rhs].Shape();
+                auto lhs_prod = std::accumulate(std::begin(lhs_shape), std::end(lhs_shape), 1,
+                                                std::multiplies<int64_t>());
+                auto rhs_prod = std::accumulate(std::begin(rhs_shape), std::end(rhs_shape), 1,
+                                                std::multiplies<int64_t>());
+                return lhs_prod > rhs_prod;
+              });
+    for (const auto& key : keys) {
+      int in_idx = graph_runtime->GetInputIndex(key);
+      if (in_idx >= 0) {
+        graph_runtime->SetInput(in_idx, const_cast<DLTensor*>(value[key].operator->()));
+      }
+    }
+  }
+
+  const std::string& GetJson() const { return graph_json_; }
```

Comment:
Likely we don't need these getters as they are only used internally (or though packedfunc)